Sources/NIOPosix/System.swift

```diff
@@ -411,14 +386,14 @@ internal enum Posix {
 
     @inline(never)
     internal static func read(descriptor: CInt, pointer: UnsafeMutableRawPointer, size: size_t) throws -> IOResult<ssize_t> {
-        return try syscall(blocking: true) {
+        return try syscall(blocking: true, unacceptableErrnoCodes: defaultUnacceptableErrnoCodes + [EINVAL]) {
```

Comment:
Unfortunately I don't think this approach is sufficient for us.

The biggest issue I have here is that this causes a runtime Array allocation on all codepaths through this function. I suspect the performance impact of this is fairly substantial.

Unfortunately, the only really effective solution at the moment is to duplicate the `isUnacceptableErrno` function with a new body. This is what we did for `close`, and sadly is what we have to do here as well.

Reply:
Ah, yup, in fact the allocation counter tests failed.